### ISSUE TITLE
evaluate SurfaceViewControlHost for GUI integration

### ISSUE THREAD

**Issue by atsushieno** [atsushieno/aap-core#157]
This looks quite interesting. I was totally unaware of it. https://commonsware.com/blog/2020/03/27/peek-surfacecontrolviewhost-android-r.html  (not likely just me; I quickly searched it on handful of communities and got no result)

**Comment by atsushieno** [comment]
I have been experimenting this SurfaceControlViewHost thing. I got basic ComposeView working in a standalone example project without any AAP context. Probably we could adopt it as an UI option. It was not simple though: I had to explore some lifecycle and state management bits and stuck at UI theme differences.

There could be more pitfalls as users explore this SurfaceControlViewHost API, but it is the best suited GUI alternative we could choose. No dangerous permissions involved. No complicated transition between UI and non-UI-yet-non-DSP code (e.g. preset loader) required.

Yet, it involves another Service which brings in some consideration. We basically avoid another connection between plugin and host, claiming that any extension requests and replies could be represented "either in the MIDI2 messages or the extension() binder messages". If there is another service connection for GUI, then the overall design might feel awkward.

Unifying the GUI service with `AudioPluginService` would be an option, but since we have to send a `Binder`, it will need another function definition which may break service compatibility (V3 -> V4).

Hopefully adding a new message definition would not break it, but we have to ensure that.
